Extensions/IISWebAppDeploy/Src/Tasks/SqlDacpacDeploy/DeployToSqlServer.ps1

```diff
@@ -72,24 +72,31 @@ function GetScriptToRun
     $serverName = EscapeSpecialChars -str $serverName
     $databaseName = EscapeSpecialChars -str $databaseName
 
+    if($sqlUserName -and $sqlPassword)
+    {
+        $securePasswordScript = "`$secureAdminPassword = ConvertTo-SecureString `"$sqlPassword`"  -AsPlainText -Force"
+        $psCredentialCreationScript = "`$sqlServerCredentials = New-Object System.Management.Automation.PSCredential (`"$sqlUserName`", `$secureAdminPassword)" 
+        $initScript = [string]::Format("{0} {1} {2}", $securePasswordScript, [Environment]::NewLine,  $psCredentialCreationScript)
+    }
+
     if ($taskType -eq "dacpac")
     {
-        $invokeMain = "Execute-DacpacDeployment -dacpacFile `"$dacpacFile`" -targetMethod $targetMethod -serverName `"$serverName`" -databaseName `"$databaseName`" -authscheme $authscheme -sqlUsername `"$sqlUsername`" -sqlPassword `"$sqlPassword`" -connectionString `"$connectionString`" -publishProfile `"$publishProfile`" -additionalArguments `"$additionalArguments`""
+        $invokeMain = "Execute-DacpacDeployment -dacpacFile `"$dacpacFile`" -targetMethod $targetMethod -serverName `"$serverName`" -databaseName `"$databaseName`" -authscheme $authscheme -sqlServerCredentials `$sqlServerCredentials -connectionString `"$connectionString`" -publishProfile `"$publishProfile`" -additionalArguments `"$additionalArguments`""
 
         $sqlPackageScript = Get-Content TaskModuleSqlUtility\SqlPackageOnTargetMachines.ps1 | Out-String
 
         Write-Verbose "Executing main function in SqlPackageOnTargetMachines : $invokeMain"
-        $sqlDacpacOnTargetMachinesScript = [string]::Format("{0} {1} ( {2} )", $sqlPackageScript,  [Environment]::NewLine,  $invokeMain)
+        $sqlDacpacOnTargetMachinesScript = [string]::Format("{0} {1} {2} {3} ( {4} )", $sqlPackageScript,  [Environment]::NewLine, $initScript, [Environment]::NewLine,  $invokeMain)
```

Comment:
Why do we need to construct a init script first and then adding it to another script? This becomes little complex.  can't we construct all required items here?